### PR TITLE
fix: publication deletion as request; change empty rules behavior 

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/service/PublicationService.java [hunk 1]
```diff
@@ -206,67 +206,78 @@ public Publication createPublication(ResourceDescription bucket, Publication pub
         checkSourceResources(publication);
         checkTargetResources(publication);
 
-        copySourceToReviewResources(publication);
+        // check target location access
+        boolean hasAccess = hasPublicAccess(context, ResourceDescription.fromPublicUrl(publication.getResources().get(0).getTargetUrl()));
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/service/PublicationService.java]: Check all sub folders?

File: src/main/java/com/epam/aidial/core/service/PublicationService.java [hunk 2]
```diff
@@ -524,6 +532,26 @@ private void checkTargetResources(Publication publication) {
...
+            if (!checkResource(descriptor)) {
+                throw new IllegalArgumentException("Target resource does not exist: " + descriptor.getUrl());
+            }
+        }
+    }
+
+    private void checkTargetResourceAccess(ProxyContext context, Publication publication) {
+        for (Publication.Resource resource : publication.getResources()) {
+            String url = resource.getTargetUrl();
+            if (!hasPublicAccess(context, ResourceDescription.fromPublicUrl(url))) {
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/service/PublicationService.java]: Please optimize it to check all resources at once.

File: src/main/java/com/epam/aidial/core/service/PublicationService.java [hunk 3]
```diff
@@ -505,6 +510,9 @@ private void checkSourceResources(Publication publication) {
     private void checkReviewResources(Publication publication) {
         for (Publication.Resource resource : publication.getResources()) {
             String url = resource.getReviewUrl();
+            if (url == null) {
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/service/PublicationService.java]: You do not need to check it. It is always not null for Publication.PENDING.

File: src/main/java/com/epam/aidial/core/service/PublicationService.java (hunk 1 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/service/PublicationService.java]: fixed

File: src/main/java/com/epam/aidial/core/service/PublicationService.java (hunk 2 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/service/PublicationService.java]: added

File: src/main/java/com/epam/aidial/core/service/PublicationService.java (hunk 3 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/service/PublicationService.java]: removed

**Review by artsiomkorzun — APPROVED** [review]

**Pull request merged**